src/Raven.Server/Documents/TransactionCommands/JsonPatchCommand.cs

```diff
@@ -0,0 +1,372 @@
+                _patchResult = new PatchResult
+                {
+                    Status = PatchStatus.NotModified,
+                };
+            }
+            else
+            {
+                _patchResult = new PatchResult
+                {
+                    Status = PatchStatus.Patched,
+                    ModifiedDocument = document.Data?.Clone(_externalContext),
+                    ChangeVector = putResult.Value.ChangeVector,
+                    Collection = putResult.Value.Collection.Name,
+                    LastModified = putResult.Value.LastModified
+                };
+            }
+
+            return 1;
+        }
+
+        public string HandleReply(DynamicJsonArray reply, HashSet<string> modifiedCollections, DocumentDatabase database)
+        {
+            if (_patchResult.ModifiedDocument != null)
+                database.HugeDocuments.AddIfDocIsHuge(_id, _patchResult.ModifiedDocument.Size);
+            
+            if (_patchResult.Collection != null)
+                modifiedCollections?.Add(_patchResult.Collection);
+
+            var patchReply = new DynamicJsonValue
+            {
+                [nameof(BatchRequestParser.CommandData.Id)] = _id,
+                [nameof(BatchRequestParser.CommandData.ChangeVector)] = _patchResult.ChangeVector,
+                [nameof(Constants.Documents.Metadata.LastModified)] = _patchResult.LastModified,
+                [nameof(BatchRequestParser.CommandData.Type)] = nameof(CommandType.JsonPatch),
+                [nameof(PatchStatus)] = _patchResult.Status
+            };
+
+            if (_returnDocument)
+                patchReply[nameof(PatchResult.ModifiedDocument)] = _patchResult.ModifiedDocument;
+
+            reply.Add(patchReply);
+            
+            return _patchResult.ChangeVector;
+        }
+        
+        public record Command(CommandTypes Type, string[] Paths, string[] FromPaths, string OriginalPath, string OriginalFrom, object Value)
+        {
+            public BlittableJsonReaderBase GetActionableObject(Document document, string[] paths, string fullPath)
+            {
+                BlittableJsonReaderBase root = document.Data;
+                
+                for (int i = 1; i < paths.Length - 1; i++)
+                {
+                    TryGetMember(root, paths[i], out var member, fullPath);
+                    
+                    if (member is BlittableJsonReaderObject nested)
+                        root = nested;
+                    else if (member is BlittableJsonReaderArray arr)
+                    {
+                        if(i == paths.Length - 2)
+                            return arr;
+                        root = arr;
+                    }
+                    else
+                        throw new ArgumentException($"Cannot reach target location. Failed to fetch '{paths[i+1]}' in path '{fullPath}' for operation '{Type}' because element at {paths[i]} is not an object or a collection");
```

Comment:
You are not specifying what is the document id here.
Better to wrap the whole calling function with a `try/catch` that would add such details